### PR TITLE
[DOC] #18228 Add example to EmberArray.forEach documentation

### PR DESCRIPTION
**Pull request by Frozenfire92**
#18228

Adds an example for the EmberArray.forEachmethod

### CODE REVIEW

File: packages/@ember/-internals/runtime/lib/mixins/array.js [hunk 1]
```diff
@@ -658,6 +658,15 @@ const ArrayMixin = Mixin.create(Enumerable, {
     object that will be set as `this` on the context. This is a good way
     to give your iterator function access to the current object.
 
+    Example Usage:
+
+    ```javascript
+      let foods = ['apple', 'banana', carrot'];
+
+      foods.forEach(food => console.log(`You should eat a ${food}`));
```
Comment (locks) [inline packages/@ember/-internals/runtime/lib/mixins/array.js]: Hm, I think we should come up with an example that does some side effect in the `forEach` block, like setting a property in the items of the array.
What if each fruit was an object, and we had an `eaten` property that we toggle to true?

**Review by locks — CHANGES_REQUESTED** [review]

File: packages/@ember/-internals/runtime/lib/mixins/array.js [hunk 2]
```diff
@@ -658,6 +658,23 @@ const ArrayMixin = Mixin.create(Enumerable, {
...
+      { name: 'apple', eaten: false },
+      { name: 'banana', eaten: false },
+      { name: carrot': eaten: false }
+    ];
+
+    foods.forEach((food) => food.eaten = true);
+
+    let output = '';
+    foods.forEach((item, index, array) => 
+      output += `${index + 1}/${array.length}) ${item}\n`
```
Comment (locks) [inline packages/@ember/-internals/runtime/lib/mixins/array.js]:
```suggestion
      output += `${index + 1}/${array.length} ${item.name}\n`;
```

File: packages/@ember/-internals/runtime/lib/mixins/array.js [hunk 3]
```diff
@@ -658,6 +658,23 @@ const ArrayMixin = Mixin.create(Enumerable, {
     object that will be set as `this` on the context. This is a good way
     to give your iterator function access to the current object.
 
+    Example Usage:
+
+    ```javascript
+    let foods = [
+      { name: 'apple', eaten: false },
+      { name: 'banana', eaten: false },
+      { name: carrot': eaten: false }
```
Comment (locks) [inline packages/@ember/-internals/runtime/lib/mixins/array.js]:
```suggestion
      { name: 'carrot', eaten: false }
```

File: packages/@ember/-internals/runtime/lib/mixins/array.js [hunk 4]
```diff
@@ -658,6 +658,23 @@ const ArrayMixin = Mixin.create(Enumerable, {
...
+    ];
+
+    foods.forEach((food) => food.eaten = true);
+
+    let output = '';
+    foods.forEach((item, index, array) => 
+      output += `${index + 1}/${array.length}) ${item}\n`
+    );
```
Comment (locks) [inline packages/@ember/-internals/runtime/lib/mixins/array.js]:
```suggestion
    );
    console.log(output);
    // 1/3 apple
    // 2/3 banana
    // 3/3 carrot
```

**Review by locks — DISMISSED** [review]

**Comment by locks** [comment]
Left some suggestions :)

**Review by pzuraq — APPROVED** [review]
It looks like the API docs currently use native arrays for a lot of examples. I think this is not ideal, since `EmberArray` is really separate from native arrays (and we want to move more that direction), so in the future we should do a sweep and convert all examples to explicitly use `Ember.A()`, but for now I think this is fine.

**Pull request merged**